tidy3d/components/geometry/base.py

```diff
@@ -272,6 +272,15 @@ def bounding_box(self):
         """
         return Box.from_bounds(*self.bounds)
 
+    @cached_property
+    def zero_dims(self) -> List[Axis]:
+        """A list of axes along which the :class:`Geometry` is zero-sized based on its bounds."""
+        zero_dims = []
+        for dim in range(3):
+            if self.bounds[1][dim] - self.bounds[0][dim] == 0:
```

Comment:
see my note, consider replacing with
```py
if self.bounds[1][dim] == self.bounds[0][dim]:
```

Reply:
Performance for this validator is no longer an issue, but I can do the modification anyway.